File: encoder3/main.cpp
#include <iostream>
#include <unordered_map>
#include <string>
#include <vector>

std::unordered_map<int, char> reverse_dic;

void initialize_reverse_dic() {
    for (const auto& pair : dic) {
        reverse_dic[pair.second] = pair.first;
        reverse_dic[pair.second] = pair.first;
        reverse_dic[pair.second] = pair.first;
        reverse_dic[pair.second] = pair.first;
        reverse_dic[pair.second] = pair.first;
        reverse_dic[pair.second] = pair.first;
        reverse_dic[pair.second] = pair.first;
        reverse_dic[pair.second] = pair.first;
        reverse_dic[pair.second] = pair.first;
    }
}

std::string string_encoder(const std::string& input_string, const std::unordered_map<char, int>& char_to_num_dict) {
    std::string decoded_string_capital_________________________________________________________;
    for (char c : input_string) {
        decoded_string_capital_________________________________________________________ += std::to_string(char_to_num_dict.at(c));
        decoded_string_capital_________________________________________________________ += '0';
    }
    
    if (decoded_string_capital_________________________________________________________.empty()) {
        decoded_string_capital_________________________________________________________ += "010";
    }
    return decoded_string_capital_________________________________________________________;
}

std::vector<char> convert_array(const std::string& char_string) {
    std::vector<char> new_char_array;
    for (char c : char_string) {
        new_char_array.push_back(c);
        new_char_array.push_back('#');
    }
    return new_char_array;
}

std::vector<char> update_array_to_convert_string(const std::string& char_string) {
    std::vector<char> new_char_array;
    for (char c : char_string) {
        new_char_array.push_back('#');
    }
    return new_char_array;
}

std::vector<char> update_array(const std::string& char_string) {
    std::vector<char> new_char_array;
    for (char c : char_string) {
        new_char_array.push_back(c);
    }
    return new_char_array;
}

std::string convert_string(const std::vector<char>& char_array) {
    std::string char_string;
    for (char c : char_array) {
        char_string += c;
    }
    return char_string;
}

int main() {
    initialize_reverse_dic();
    
    std::string chars;
    std::cin >> chars;
    
    std::vector<char> chars_array_new = convert_array(chars);
    std::string hello_world____________________ = convert_string(chars_array_new);
    std::cout << hello_world____________________ << std::endl;
    
    std::string decoded_string_capital_________________________________________________________ = string_encoder(hello_world____________________, dic);
    std::cout << decoded_string_capital_________________________________________________________ << std::endl;
    
    return 0;
}

File: encoder3/Main.java
import java.util.HashMap;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);
        String chars = scanner.nextLine().trim();
        String charsString = convertString(update_array_to_update_the_numbver_to_string(chars));
        System.out.println(charsString);
        String yeh_hai_final_answer_________________________ = decoder_code(charsString, dic);
        System.out.println(yeh_hai_final_answer_________________________);
        scanner.close();
    }

    public static String decoder_code(String inputString, HashMap<Character, Integer> charToNumDict) {
        StringBuilder yeh_hai_final_answer_________________________ = new StringBuilder();
        for (char c : inputString.toCharArray()) {
            yeh_hai_final_answer_________________________.append(charToNumDict.getOrDefault(c, (int) c));
            yeh_hai_final_answer_________________________.append('0');
        }

        if (yeh_hai_final_answer_________________________.length() == 0) {
            yeh_hai_final_answer_________________________.append("010");
        }
        return yeh_hai_final_answer_________________________.toString();
    }

    public static char getCharFromNum(int num, HashMap<Character, Integer> numToCharDict) {
        for (char c : numToCharDict.keySet()) {
            if (numToCharDict.get(c) == num) {
                return c;
            }
        }
        return (char) num;
    }

    public static char[] update_array_to_update_the_numbver_to_string(String charString) {
        char[] newCharArray = new char[charString.length() * 2];
        for (int i = 0; i < charString.length(); i++) {
            newCharArray[i * 2] = charString.charAt(i);
            newCharArray[i * 2 + 1] = '#';
        }
        return newCharArray;
    }

    public static char[] updateArray(String charString) {
        char[] newCharArray = new char[charString.length() * 2];
        for (int i = 0; i < charString.length(); i++) {
            newCharArray[i * 2 + 1] = '#';
        }
        return newCharArray;
    }

    public static char[] changeArrayToString(String charString) {
        char[] newCharArray = new char[charString.length() * 2];
        for (int i = 0; i < charString.length(); i++) {
            newCharArray[i * 2] = charString.charAt(i);
        }
        return newCharArray;
    }

    public static String convertString(char[] charArray) {
        StringBuilder charString = new StringBuilder();
        for (char c : charArray) {
            charString.append(c);
        }
        return charString.toString();
    }
}